src/lib.rs

```diff
@@ -695,6 +696,14 @@ pub trait ElvenTools {
         left_tokens
     }
 
+    // Get the total amount of tokens
+    #[view(getTotalTokens)]
+    fn total_tokens(&self) -> u32 {
```

Comment:
I think we don't need this separate function. We can leave the view only on the storage as it is below